src/transformers/utils/generic.py

```diff
@@ -71,31 +71,63 @@ def strtobool(val):
     raise ValueError(f"invalid truth value {val!r}")
 
 
+def infer_framework_from_repr(x):
+    """
+    Tries to guess the framework of an object `x` from its repr (brittle but will help in `is_tensor` to try the
+    frameworks in a smart order).
+    """
+    representation = repr(x)
+    if representation.startswith("tensor"):
+        return "pt"
+    elif "tf.Tensor" in representation:
+        return "tf"
+    elif representation.startswith("Array"):
+        return "jax"
+    elif representation.startswith("array"):
+        return "np"
+
+
+def _get_frameworks_and_tests(x):
```

Comment:
For me, I would love to name something like `_get_frameworks_to_is_tensor_fn` or even `_frameworks_to_is_tensor_fn``